otter/cloud_client.py

```diff
@@ -507,6 +524,25 @@ class NovaComputeFaultError(Exception):
     """
 
 
+@attributes([])
```

Comment:
Oh, no, right, it's so that characteristic can set up the equality test for me.  Otherwise passing a unicode message vs a bytes message would mean that that the errors were different.

This actually brings up a question I had about characteristic vs attr btw - I was thinking about converting these to use `@attr.s`, but attr doesn't pass extra arguments through to the `__init__`, and characteristic does.

What that means is that we can't extend `Exceptions` and be able to use `Exception("msg")`.  We *can* define our own `message` attribute to our exceptions.  Wondering if that would be better?